src/layerstack.stanza

```diff
@@ -236,27 +245,36 @@ public defn LayerSpec (
 
 doc: \<DOC>
 Create a Copper layer
+  Usage: Copper(0.3) : specify thickness 
+         Copper(0.3, name = "Cu") : specify thickness and name
+         Copper(0.3, AluminumMaterial, name = "Cu") : specify thickness, material and name
+@param thickness thickness of the layer in mm - required argument
+@param material material of the layer - optional positional argument
+@param name name of the layer - optional named argument
 <DOC>
-public defn Copper (name:String, weight:Double, material:ConductorMaterial = CopperMaterial) -> LayerSpec :
-  LayerSpec(name = name, thickness = compute-thickness(material, weight), material = material)
-
-public defn Copper (weight:Double, material:ConductorMaterial = CopperMaterial) -> LayerSpec :
-  LayerSpec(thickness = compute-thickness(material, weight), material = material)
-
-public defn FR4 (name:String, thickness:Double, material:DielectricMaterial = FR4-Material) -> LayerSpec :
-  LayerSpec(name = name, material = material, thickness = thickness)
-
-public defn FR4 (thickness:Double, material:DielectricMaterial = FR4-Material) -> LayerSpec :
-  LayerSpec(material = material, thickness = thickness)
-
-public defn Soldermask (name:String, thickness:Double, material:DielectricMaterial = SoldermaskMaterial) -> LayerSpec :
-  LayerSpec(name = name, material = material, thickness = thickness)
-
-public defn Soldermask (thickness:Double, material:DielectricMaterial = SoldermaskMaterial) -> LayerSpec :
-  LayerSpec(material = material, thickness = thickness)
-
-defn get-name (x:LayerSpec) -> Maybe<String> :
-  name(x)
+public defn Copper (thickness:Double, material:ConductorMaterial = CopperMaterial
+    -- name:String = ?) -> LayerSpec :
+  match(name):
+    (name:One) :
+      LayerSpec(name = value(name), thickness = thickness, material = material)
+    (_):
+      LayerSpec(thickness = thickness, material = material)
+
+public defn FR4 (thickness:Double, material:DielectricMaterial = FR4-Material
+    -- name:String = ?) -> LayerSpec :
+  match(name):
+    (name:One) :
+      LayerSpec(name = value(name), thickness = thickness, material = material)
+    (_):
+      LayerSpec(thickness = thickness, material = material)
+
+public defn Soldermask (thickness:Double, material:DielectricMaterial = SoldermaskMaterial
+    -- name:String = ?) -> LayerSpec :
+  match(name):
+    (name:One) :
+      LayerSpec(name = value(name), thickness = thickness, material = material)
+    (_):
+      LayerSpec(thickness = thickness, material = material)
```

Comment:
I understand that you did this with a `#for` and so there wasn't duplication but not that it is unrolled there is duplication. 

Can we please refactor these methods to use the `#LayerSpec` underlying constructor so that we can avoid the need to have two versions of the constructor call ? 

```
public defn Soldermask (thickness:Double, material:DielectricMaterial = SoldermaskMaterial
    -- name:String = ?) -> LayerSpec :
      #LayerSpec(name, None(), material, thickness)
```

Can we also put another optional keyword for `description:String = ?` so that the user can override that if they care to ?